cocotb/share/include/gpi_logging.h

```diff
@@ -30,39 +30,183 @@
 #ifndef COCOTB_GPI_LOGGING_H_
 #define COCOTB_GPI_LOGGING_H_
 
+#include <cstdarg>
+
 #ifdef __cplusplus
-# define EXTERN_C_START extern "C" {
-# define EXTERN_C_END }
-#else
-# define EXTERN_C_START
-# define EXTERN_C_END
+extern "C" {
 #endif
 
-EXTERN_C_START
-
+/** Named logging level
+ *
+ *  Only logs at one of these log level will print a level name.
```

Comment:
This isn't true:
* The GPI log implementer could choose to print other level names
* In python, `logging.addLevelName` can be used to associate new level names

I'd probably instead say something like
```suggestion
 *  These logging levels deliberately align with the builtin constants in the python logging module.
 *  Implementers of `gpi_log_handler_type` should emit human readable level names for these value, but may support other values
```